Orange-OpenSource/its-client
# Rust examples

New features
------------------

* `geo_routing` features; provides an implementation of Topic using a quadkey to allow further subscription on a specific zone
* copycat example (closes #87)

How to test
---------------

1. Edit `config.ini` file to fit your broker's configuration
2. Run the example
    ```
    cargo run --example copycat --features="geo_routing" -- --config config.ini -u <username> -p <password>
    ```
    **=> Logs must tell about CAM reception**
    **=> After 3 seconds the example should start sending CAM messages**
    > ...  
    > ora_copycat-nbu_13101 cam received_on ora_broker_3101/5GCroCo/outQueue/v2x/cam/ora_car_sumoA320.9 12558549/18870 at 1714643529272  
    > ora_copycat-nbu_13101 cam received_on ora_broker_3101/5GCroCo/outQueue/v2x/cam/ora_car_sumoA320.2 10614122/19039 at 1714643529442  
    > ora_copycat-nbu_13101 cam sent_on ora_broker_3101/5GCroCo/inQueue/v2x/cam/ora_copycat-nbu_13101 15992436/16036 at 1714643529444  
    > ora_copycat-nbu_13101 cam received_on ora_broker_3101/5GCroCo/outQueue/v2x/cam/ora_car_sumoA320.3 9201122/19041 at 1714643529444  
    > ora_copycat-nbu_13101 cam received_on ora_broker_3101/5GCroCo/outQueue/v2x/cam/ora_car_sumoA320.4 13541139/19042 at 1714643529445  
    > ora_copycat-nbu_13101 cam sent_on ora_broker_3101/5GCroCo/inQueue/v2x/cam/ora_copycat-nbu_13101 10617223/16038 at 1714643529446  
    > ...
